shell/common/engine.h

```diff
@@ -663,6 +677,41 @@ class Engine final : public RuntimeDelegate, PointerDataDispatcher::Delegate {
   ///
   void DispatchPlatformMessage(fml::RefPtr<PlatformMessage> message);
 
+  //----------------------------------------------------------------------------
+  /// @brief      A version of `DispatchPlatformMessage` that allows delegating
+  ///             to a RuntimeControllerProxy.  You probably want the other
+  ///             `DispatchPlatformMessage` unless you are writing tests.
+  ///
+  template <typename RuntimeControllerProxy>
+  void DispatchPlatformMessage(fml::RefPtr<PlatformMessage> message,
```

Comment:
This might not scale well in the future, we'll see.  It was this or make RuntimeControllerProxy not final and give it virtual methods, which would have runtime costs.  `RuntimeController` is very complicated to use in tests since it requires a dart snapshot and everything that goes along with that.

Reply:
Also worth noting, I tried to make this method private but that doesn't scale well since googletest compiles each test as its own class that has to be friended.  I can still do that if you want.  It's not breaking any encapsulation having it public, it is a bit cleaner.